File: src/models/creature/Action.ts
import ActionType from './ActionType';
import { AttackDelivery, AttackType } from './AttackType';
import Damage from './Damage';

type Action = {
  id: number;
  name: string;
  description: string;
  isAttack: boolean;
  toHit: number | null;
  combatantsHit: number | null;
  damage: Damage[] | null;
  reach: number | null;

  actionTypeId: number | null;
  attackDeliveryId: number | null;
  attackTypeId: number | null;

  actionType: ActionType | null;
  attackDelivery: AttackDelivery | null;
  attackType: AttackType | null;
};

export default Action;

File: src-tauri/src/models/action.rs
use crate::models::damage::Damage;
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Action {
    name: String,
    description: String,
    action_type: String,
    is_attack: bool,
    attack_delivery: Option<String>,
    to_hit: Option<i16>,
    damage: Option<Vec<Damage>>,
    reach: Option<i16>,
}
